service/state/height.go

```diff
@@ -54,20 +42,9 @@ func (h *Height) ForBlock(blockID flow.Identifier) (uint64, error) {
 }
 
 func (h *Height) ForCommit(commit flow.StateCommitment) (uint64, error) {
-	key := make([]byte, 1+len(commit))
-	key[0] = prefixIndexCommit
-	copy(key[1:], commit[:])
 	var height uint64
 	err := h.core.db.View(func(tx *badger.Txn) error {
-		item, err := tx.Get(key)
-		if err != nil {
-			return fmt.Errorf("could not retrieve commit index: %w", err)
-		}
-		_ = item.Value(func(val []byte) error {
-			height = binary.BigEndian.Uint64(val)
-			return nil
-		})
-		return nil
+		return RetrieveBlockHeight(Encode(prefixIndexBlock, commit), &height)(tx)
```

Comment:
This is exactly why I would like these functions to be fully strongly typed. There is an error here, it should be `RetrieveBlockCommit` in your code. However, as mentioned before, this would become `RetrieveHeightByCommit(commit, &height)(tx)`.